core/src/main/java/io/questdb/cairo/frm/file/ContiguousFileFixFrameColumn.java

```diff
@@ -73,19 +76,41 @@ public void append(long offset, FrameColumn sourceColumn, long sourceLo, long so
                 int sourceFd = sourceColumn.getPrimaryFd();
                 long length = sourceHi << shl;
                 TableUtils.allocateDiskSpaceToPage(ff, fd, (offset + sourceHi) << shl);
-                ff.fadvise(sourceFd, sourceLo << shl, length, Files.POSIX_FADV_SEQUENTIAL);
-                if (ff.copyData(sourceFd, fd, sourceLo << shl, offset << shl, length) != length) {
-                    throw CairoException.critical(ff.errno()).put("Cannot copy data [fd=").put(fd)
-                            .put(", destOffset=").put(offset << shl)
-                            .put(", size=").put(length)
-                            .put(", fileSize=").put(ff.length(fd))
-                            .put(", srcFd=").put(sourceFd)
-                            .put(", srcOffset=").put(sourceLo << shl)
-                            .put(", srcFileSize=").put(ff.length(sourceFd))
-                            .put(']');
-                }
-                if (commitMode != CommitMode.NOSYNC) {
-                    ff.fsync(fd);
+                if (mixedIOFlag) {
+                    ff.fadvise(sourceFd, sourceLo << shl, length, Files.POSIX_FADV_SEQUENTIAL);
```

Comment:
this is column append logic, having dealt with spurious reads should we not be using FADV_RANDOM here? Also, is it worth advising the same for mmap?

Reply:
Mmap already has an madvise call - see `TableUtils#mapAppendColumnBuffer()`. As for `FADV_SEQUENTIAL`, `sourceFd` stands for a read-only file here. Do you refer to `fd`, i.e. the destination file? If so, I added the fadvise calls in dba8154b64acbd435a0489d13dc639c9a7a38269

Reply:
After some thinking, I removed these and other write-related `fadvise` calls in 0d31729d800cb996b707e36016c3539f35fd33ba

That's because they're redundant and don't change anything. Spurious reads won't happen here since we don't use mmap and, hence, there are no page faults. Also, as we make a single `sendfile64` or `pwrite` call with the fds, the kernel has all the inputs, source and dest fds and the length, so the readahead is not needed.

Reply:
spuriouis reads are not mmap specific, they are a function of buffered IO. Are they not?

Reply:
What we had is mmap specific.